vmsize_test.go

```diff
@@ -0,0 +1,192 @@
+			family:           "N",
+			subfamily:        stringToPtr("V"),
+			cpus:             "16",
+			cpusConstrained:  nil,
+			additiveFeatures: []rune{'a', 's'},
+			acceleratorType:  nil,
+			version:          "v4",
+			series:           "NVas_v4",
+		},
+	},
+	{
+		name: "Standard_M16ms",
+		size: "M16ms_v2",
+		expectedVM: VMSizeType{
+			family:           "M",
+			subfamily:        nil,
+			cpus:             "16",
+			cpusConstrained:  nil,
+			additiveFeatures: []rune{'m', 's'},
+			acceleratorType:  nil,
+			version:          "v2",
+			series:           "Mms_v2",
+		},
+	},
+	{
+		name: "Standard_NC4as_T4_v3",
+		size: "NC4as_T4_v3",
+		expectedVM: VMSizeType{
+			family:           "N",
+			subfamily:        stringToPtr("C"),
+			cpus:             "4",
+			cpusConstrained:  nil,
+			additiveFeatures: []rune{'a', 's'},
+			acceleratorType:  stringToPtr("T4"),
+			version:          "v3",
+			series:           "NCas_v3",
+		},
+	},
+	{
+		name: "Standard_M8-2ms",
+		size: "M8-2ms_v2",
+		expectedVM: VMSizeType{
+			family:           "M",
+			subfamily:        nil,
+			cpus:             "8",
+			cpusConstrained:  stringToPtr("2"),
+			additiveFeatures: []rune{'m', 's'},
+			acceleratorType:  nil,
+			version:          "v2",
+			series:           "Mms_v2",
+		},
+	},
+	{
+		name: "Standard_A4_v2",
+		size: "A4_v2",
+		expectedVM: VMSizeType{
+			family:           "A",
+			subfamily:        nil,
+			cpus:             "4",
+			cpusConstrained:  nil,
+			additiveFeatures: []rune{},
+			acceleratorType:  nil,
+			version:          "v2",
+			series:           "A_v2",
+		},
+	},
+}
+
+// Test_getSubFamily tests the getSubFamily() function.
+func Test_getSubFamily(t *testing.T) {
+	a := assert.New(t)
+	for _, test := range testCases {
+		// Test GetSubFamily()
+		subFamily, err := getSubFamily(test.size)
+		if err != nil {
+			t.Fatalf(`getSubFamily failed %s, %v`, test.size, err)
+		}
+		a.Equal(test.expectedVM.subfamily, subFamily)
+	}
+}
+
+// Test_getConstrainedVCPUs tests the getSubFamily() function.
+func Test_getConstrainedVCPUs(t *testing.T) {
+	a := assert.New(t)
+	for _, test := range testCases {
+		// Test GetConstrainedVCPUs()
+		constrainedVCPUs, err := getConstrainedVCPUs(test.size)
+		if err != nil {
+			t.Fatalf(`getConstrainedVCPUs failed %s, %v`, test.size, err)
+		}
+		a.Equal(test.expectedVM.cpusConstrained, constrainedVCPUs)
+	}
+}
+
+// Test_getAdditiveFeatures tests the getAdditiveFeatures() function.
+func Test_getAdditiveFeatures(t *testing.T) {
+	a := assert.New(t)
+	for _, test := range testCases {
+		// Test GetAdditiveFeatures()
+		additiveFeatures, err := getAdditiveFeatures(test.size)
+		if err != nil {
+			t.Fatalf(`getAdditiveFeatures failed %s, %v`, test.size, err)
+		}
+		a.Equal(test.expectedVM.additiveFeatures, additiveFeatures)
+	}
+}
+
+// Test_getAcceleratorType tests the getAcceleratorType() function.
+func Test_getAcceleratorType(t *testing.T) {
+	a := assert.New(t)
+	for _, test := range testCases {
```

Comment:
what is the advantage to having a lot of separate loops vs one main loop with many assertions for Accel type, Subfamily etc